ecosystem/typescript/sdk_v2/src/transactions/transaction_builder/transaction_builder.ts

```diff
@@ -0,0 +1,512 @@
+export function generateTransactionPayload(args: ScriptData): TransactionPayloadScript;
+export function generateTransactionPayload(args: MultiSigData): TransactionPayloadMultisig;
+export function generateTransactionPayload(args: GenerateTransactionPayloadData): TransactionPayload;
+/**
+ * Builds a transaction payload based on the data argument and returns
+ * a transaction payload - TransactionPayloadScript | TransactionPayloadMultisig | TransactionPayloadEntryFunction
+ *
+ * @param args.data GenerateTransactionPayloadData
+ *
+ * @return TransactionPayload
+ */
+export function generateTransactionPayload(args: GenerateTransactionPayloadData): TransactionPayload {
+  // generate script payload
+  if ("bytecode" in args) {
+    const scriptPayload = new TransactionPayloadScript(
+      new Script(hexToBytes(args.bytecode), args.type_arguments, args.arguments),
+    );
+    return scriptPayload;
+  }
+
+  // generate multi sig payload
+  if ("multisigAddress" in args) {
+    const funcNameParts = args.function.split("::");
+    const multiSigPayload = new TransactionPayloadMultisig(
+      new MultiSig(
+        args.multisigAddress,
+        new MultiSigTransactionPayload(
+          EntryFunction.build(
+            `${funcNameParts[0]}::${funcNameParts[1]}`,
+            funcNameParts[2],
+            args.type_arguments,
+            args.arguments,
+          ),
+        ),
+      ),
+    );
+    return multiSigPayload;
+  }
+
+  // generate entry function payload
+  const funcNameParts = args.function.split("::");
+  const entryFunctionPayload = new TransactionPayloadEntryFunction(
+    EntryFunction.build(
+      `${funcNameParts[0]}::${funcNameParts[1]}`,
+      funcNameParts[2],
+      args.type_arguments,
+      args.arguments,
+    ),
+  );
+  return entryFunctionPayload;
+}
+
+/**
+ * Generates a raw transaction
+ *
+ * @param args.aptosConfig AptosConfig
+ * @param args.sendet The transaction's sender account address as a hex input
+ * @param args.payload The transaction payload - can create by using generateTransactionPayload()
+ *
+ * @returns RawTransaction
+ */
+export async function generateRawTransaction(args: {
+  aptosConfig: AptosConfig;
+  sender: HexInput;
+  payload: TransactionPayload;
+  options?: GenerateTransactionOptions;
+}): Promise<RawTransaction> {
+  const { aptosConfig, sender, payload, options } = args;
+  const [{ sequence_number: sequenceNumber }, chainId, { gas_estimate: gasEstimate }] = await Promise.all([
+    options?.accountSequenceNumber
+      ? Promise.resolve({ sequence_number: options.accountSequenceNumber })
+      : await getInfo({ aptosConfig, accountAddress: sender }),
+    NetworkToChainId[aptosConfig.network] ?? (await getLedgerInfo({ aptosConfig })).chain_id,
+    options?.gasUnitPrice
+      ? Promise.resolve({ gas_estimate: options.gasUnitPrice })
+      : await getGasPriceEstimation({ aptosConfig }),
+  ]);
```

Comment:
Is there a general way to break this up so it's not a 10 line ternary wrapped line.  It's fairly hard to read.